### PR TITLE
Fix/guest auth 429

### PR DESCRIPTION
**Pull request by samderanova**
Addresses #147.

### CODE REVIEW

**Review by rileysw — APPROVED** [review]
Code looks good, did you test and try to reproduce the error?

**Comment by IanWearsHat** [comment]
Logging in twice as guest still raises the exception for me, although it is the updated message now.

File: apps/api/tests/test_guest.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def test_requesting_login_when_previous_key_exists_causes_429(
 
 
 @patch("auth.guest_auth._get_existing_key", autospec=True)
+@patch("auth.guest_auth._remove_guest_key", autospec=True)
 def test_successful_guest_verification_provides_identity(
+    mock_remove_guest_key: AsyncMock,
     mock_get_existing_key: AsyncMock,
```
Comment (taesungh) [inline apps/api/tests/test_guest.py:91]: Suggestion [repeating]: patch order should usually reflect order of usage/calls during test.

File: apps/api/tests/test_guest.py [hunk 2]
```diff
@@ -105,7 +107,9 @@ def test_successful_guest_verification_provides_identity(
 
 
 @patch("auth.guest_auth._get_existing_key", autospec=True)
+@patch("auth.guest_auth._remove_guest_key", autospec=True)
 def test_invalid_guest_verification_is_unauthorized(
+    mock_remove_guest_key: AsyncMock,
```
Comment (taesungh) [inline apps/api/tests/test_guest.py:112]: Suggestion [repeating]: the mock is currently unused in the function. We could call the `assert_awaited_once_with` method to further validate its usage.

**Review by taesungh — APPROVED** [review]
Simple enough, thanks for figuring out how to work with the unit tests.

**Comment by samderanova** [comment]
I tested it on the most recent preview and it works for me.

**Comment by IanWearsHat** [comment]
> Logging in twice as guest still raises the exception for me, although it is the updated message now.

Logging in, logging out, and logging back in works.

Inputting an email, backing out, and then inputting the same email triggers the exception, although I don't know if this is intended behavior.

**Comment by taesungh** [comment]
> Inputting an email, backing out, and then inputting the same email triggers the exception, although I don't know if this is intended behavior.

Yes that is intended because we don't want someone who started the process to be able to spam us, especially when each initiation causes an email to be sent.

**Review by IanWearsHat — APPROVED** [review]
Looks good then

**Pull request merged**